### PR TITLE
Adding missing attributes in for ThirdPartySync PULL

### PR DESCRIPTION
**Pull request by comunacho**
The `AndroidStringDocumentMapper` instance we are using in the `ThirdPartySync` PULL action is missing some attributes.

### CODE REVIEW

File: webapp/src/main/java/com/box/l10n/mojito/service/thirdparty/smartling/SmartlingPluralFix.java [hunk 1]
```diff
@@ -25,6 +26,19 @@ private SmartlingPluralFix() {
                     tu.setPluralForm("many");
                     tu.setPluralFormOther(textUnit.getPluralFormOther());
                     tu.setTarget(textUnit.getTarget());
+
+                    if (!Strings.isNullOrEmpty(textUnit.getRepositoryName())){
```
Comment (aurambaj) [inline webapp/src/main/java/com/box/l10n/mojito/service/thirdparty/smartling/SmartlingPluralFix.java:30]: why do you need to test for null or empty for those 3?

File: webapp/src/main/java/com/box/l10n/mojito/service/thirdparty/ThirdPartyTMSSmartling.java [hunk 2]
```diff
@@ -223,25 +223,26 @@ public void pull(Repository repository,
                      List<String> optionList) {
 
         SmartlingOptions options = SmartlingOptions.parseList(optionList);
-        AndroidStringDocumentMapper mapper = new AndroidStringDocumentMapper(pluralSeparator, null);
```
Comment (aurambaj) [inline webapp/src/main/java/com/box/l10n/mojito/service/thirdparty/ThirdPartyTMSSmartling.java:226]: can matching ctor be removed?

File: webapp/src/main/java/com/box/l10n/mojito/service/thirdparty/smartling/SmartlingPluralFix.java (hunk 1 above)
Comment (comunacho) [reply webapp/src/main/java/com/box/l10n/mojito/service/thirdparty/smartling/SmartlingPluralFix.java:30]: I just follow this pattern with strings that can be empty or null, but also there's no point in doing the assignation if is either empty or null, right?

File: webapp/src/main/java/com/box/l10n/mojito/service/thirdparty/ThirdPartyTMSSmartling.java (hunk 2 above)
Comment (comunacho) [reply webapp/src/main/java/com/box/l10n/mojito/service/thirdparty/ThirdPartyTMSSmartling.java:226]: We can't, since its being used for other use cases.

File: webapp/src/main/java/com/box/l10n/mojito/service/thirdparty/smartling/SmartlingPluralFix.java (hunk 1 above)
Comment (aurambaj) [reply webapp/src/main/java/com/box/l10n/mojito/service/thirdparty/smartling/SmartlingPluralFix.java:30]: everywhere else for pojo mapping we don't do that, so let's keep it consistent with other places. Side notes those (repository name, locale, path)shouldn't be empty because it doesn't make sense though code wise there are little checks.

Comment (comunacho) [reply webapp/src/main/java/com/box/l10n/mojito/service/thirdparty/smartling/SmartlingPluralFix.java]: Updated

File: webapp/src/main/java/com/box/l10n/mojito/service/thirdparty/ThirdPartyTMSSmartling.java (hunk 2 above)
Comment (aurambaj) [reply webapp/src/main/java/com/box/l10n/mojito/service/thirdparty/ThirdPartyTMSSmartling.java:226]: not sure in what case we don't have the repository name and locale name, even i'm confused why it is fine not having it. in fact relooking at the class I'm not sure why that class is not a stateless component?

Comment (comunacho) [reply webapp/src/main/java/com/box/l10n/mojito/service/thirdparty/ThirdPartyTMSSmartling.java:226]: When we do `List<TextUnitDTO>` to `AndroidStringDocument` we don't need the repository or targetLocale, we need it the other way around. But the class isn't stateful, each call to `AndroidStringDocumentMapper.mapToTextUnits` is self-contained.

Comment (aurambaj) [reply webapp/src/main/java/com/box/l10n/mojito/service/thirdparty/ThirdPartyTMSSmartling.java:226]: right, so it looks like a component would make more sense, and pass the required parameters to each function would make it more clearer what is needed.
